src/lib/decorators.ts

```diff
@@ -465,9 +465,10 @@ export function queryAssignedNodes(
         if (nodes && selector) {
           nodes = nodes.filter(
               (node) => node.nodeType === Node.ELEMENT_NODE &&
-                      (node as Element).matches ?
-                  (node as Element).matches(selector) :
-                  legacyMatches.call(node as Element, selector));
+                  // tslint:disable-next-line:no-any testing existence on older browsers
+                  ((node as any).matches ?
```

Comment:
Still curious about this cast here. Why is it necessary?

Reply:
TypeScript complains that the check is unnecessary because`matches` always exists. The previous code probably didn't analyze path.